python/cudaq/_cudaq.cpp

```diff
@@ -53,6 +56,19 @@ PYBIND11_MODULE(_pycudaq, mod) {
   mpiSubmodule.def(
       "initialize", []() { cudaq::mpi::initialize(); },
       "Initialize MPI if available.");
+  mpiSubmodule.def(
+      "rank", []() { return cudaq::mpi::rank(); }, "");
+  mpiSubmodule.def(
+      "num_ranks", []() { return cudaq::mpi::num_ranks(); }, "");
+  mpiSubmodule.def(
+      "all_gather",
```

Comment:
Do we have an example using this?

Reply:
I do, but the original MPI tests I had were removed before the release.

Reply:
I can add one